src/store.cc

```diff
@@ -2029,13 +2029,24 @@ StoreEntry::detachFromDisk()
 void
 StoreEntry::checkDisk() const
 {
-    const bool ok = (swap_dirn < 0) == (swap_filen < 0) &&
-                    (swap_dirn < 0) == (swap_status == SWAPOUT_NONE) &&
-                    (swap_dirn < 0 || swap_dirn < Config.cacheSwap.n_configured);
-
-    if (!ok) {
+    try {
+        Must((swap_dirn < 0) == (swap_filen < 0));
```

Comment:
Let's simplify this condition by checking the same underlying principle inside each part of the `if` statement below.